lib/bulk/unordered.js

```diff
@@ -445,12 +447,26 @@ var executeBatch = function(self, batch, options, callback) {
 
   var resultHandler = function(err, result) {
     // Error is a driver related error not a bulk op error, terminate
-    if ((err && err.driver) || (err && err.message)) {
+    if (((err && err.driver) || (err && err.message)) && !(err instanceof MongoWriteConcernError)) {
       return handleCallback(callback, err);
     }
 
     // If we have and error
     if (err) err.ok = 0;
+    if (err instanceof MongoWriteConcernError) {
```

Comment:
This seems like a lot of duplicated logic. Can we refactor the functions to be a little more DRY?